test/util/framework/hcp_helper.go

```diff
@@ -200,6 +193,28 @@ func DeleteAllHCPClusters(
 	return nil
 }
 
+func ListHCPClusterNames(
```

Comment:
This is a tricky type of utility to write - as the consumers of it need more of the underlying structure, you end up with a very leaky signature or you end up with a very shallow wrapper around the pager. Doesn't the `Azure/azure-sdk-for-go` provide a generic implementation of "collect all items in a pager" you could use instead?